src/ParseObject.js

```diff
@@ -131,9 +132,10 @@ export default class ParseObject {
   /** Prototype getters / setters **/
 
   get attributes(): AttributeMap {
-    return Object.freeze(
-      ObjectState.estimateAttributes(this.className, this._getStateIdentifier())
-    );
+    let attributes = singleInstance ?
```

Comment:
I think it would be better if we abstract this into `InstanceStateController` that can be configured by `CoreManager.js`